QuestionAnswering/QA_Agent/deployment/fastapi_app/services/nlq2action/execute_action/func/sg_carpark.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import time
+from typing import Annotated, List
+
+from fastapi import Depends
+
+from services.support_data import QAStep
+from core.kg import KgClient
+from utils.rdf import flatten_sparql_response
+from services.kg import get_sgCarpark_bgClient
+from services.feature_info_client import FeatureInfoClient, get_featureInfoClient
+from services.geocoding import IGeocoder, get_geocoder
+from services.support_data import TableDataItem
+from services.nlq2action.execute_action.func.base import Name2Func
+
+logger = logging.getLogger(__name__)
+
+
+class SGCarParkFuncExecutor(Name2Func):
+    def __init__(
+        self,
+        bg_client: KgClient,
+        feature_info_client: FeatureInfoClient,
+        geocoder: IGeocoder,
+    ):
+        self.bg_client = bg_client
+        self.feature_info_client = feature_info_client
+        self.geocoder = geocoder
+
+    def get_name2func(self):
+        return {"find_nearest_carpark": self.find_nearest_carpark}
+
+    def find_nearest_carpark(self, location: str):
+        steps: List[QAStep] = []
+
+        logger.info("Get coordinates for the location: " + location)
+        timestamp = time.time()
+        place = self.geocoder.search(location)
+        latency = time.time() - timestamp
+        logger.info("Geo-decoded data: " + str(place))
+        steps.append(
+            QAStep(
+                action="geodecode",
+                arguments=location,
+                results=str(place),
+                latency=latency,
+            )
+        )
+
+        timestamp = time.time()
+        query = """PREFIX carpark:	<https://www.theworldavatar.com/kg/ontocarpark/>
+    PREFIX geo: <http://www.opengis.net/ont/geosparql#>
+    PREFIX geof: <http://www.opengis.net/def/function/geosparql/>
+    PREFIX unit: <http://qudt.org/vocab/unit/>
+            
+    SELECT * WHERE {{
+    ?Carpark a carpark:Carpark; rdfs:label ?Label .
+    SERVICE <http://sg-ontop:8080/sparql> {{
+        SELECT (geof:distance(?Coords, "<http://www.opengis.net/def/crs/OGC/1.3/CRS84> POINT({lon} {lat})"^^geo:wktLiteral, <http://www.opengis.net/def/uom/OGC/1.0/metre>) as ?Distance) ?Carpark ?Coords
```

Comment:
In case you didn't know `<http://www.opengis.net/def/crs/OGC/1.3/CRS84>` is the default CRS with GeoSPARQL so shouldn't be needed here. Saying that, versions of Ontop before version 5, which the stack now uses, might have needed this.

Reply:
Hi @kok-foong, could you help confirm if `<http://www.opengis.net/def/crs/OGC/1.3/CRS84>` is needed here for the query that finds the carpark nearest to a give set of coordinates, in view of the current setup and future compatibility on your end?

Reply:
@kok-foong, please check my recommendation in [this comment](https://github.com/cambridge-cares/TheWorldAvatar/pull/1190#discussion_r1592535569).